File: python/dsa/data-structures/linked-list/linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def search(self, head: ListNode, target: int) -> bool:
        current = head
        while current:
            if current.val == target:
                return True
            current = current.next
        return False

File: go/dsa/data-structures/linked-list/linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func search(head *ListNode, target int) bool {
    current := head
    for current != nil {
        if current.Val == target {
            return true
        }
        current = current.Next
    }
    return false
}